src/rt/sched/cown.h

```diff
@@ -771,11 +775,67 @@ namespace verona::rt
         Scheduler::record_inflight_message();
       }
 
+      if ((sched != nullptr) && (sched->message_body != nullptr))
+        backpressure_scan(*sched->message_body, *body);
+
       // Try to acquire as many cowns as possible without rescheduling,
       // starting from the beginning.
       fast_send(body, epoch);
     }
 
+    static void
+    backpressure_scan(const MessageBody& senders, const MessageBody& receivers)
+    {
+      uint32_t pressure = 0;
+      for (size_t r = 0; r < receivers.count; r++)
+      {
+        auto* receiver = receivers.cowns[r];
+        const auto bp = receiver->backpressure.load(std::memory_order_acquire);
+        const auto load = bp.total_load();
+        pressure += load >> 3;
+        if (load > Backpressure::overload_threshold)
+          Scheduler::local()->overload(receiver);
+      }
+
+      for (size_t s = 0; s < senders.count; s++)
+      {
+        auto* sender = senders.cowns[s];
+        auto bp = sender->backpressure.load(std::memory_order_relaxed);
+        bp.pressure_add(pressure);
+        sender->backpressure.store(bp, std::memory_order_release);
+      }
+    }
+
+    bool apply_backpressure(Alloc* alloc, MessageBody* senders)
+    {
+      bool muted = false;
+      const auto bp0 =
+        senders->cowns[0]->backpressure.load(std::memory_order_relaxed);
```

Comment:
The load on the first cown is tracked normally, however the other participants often have their load reset whenever they are acquired for the message. This means that, for example, the receivers in `perf-cowngc2` all give off a load signal that is too weak to trigger adequate backpressure since the receivers are constantly participating in different multimessages and shuffling around which one is at index 0.

Reply:
So the load is distributed from the first cown.